apps/app_rpt/rpt_config.c

```diff
@@ -373,53 +367,133 @@ static int node_lookup_bydns(const char *node, char *nodedata, size_t nodedatale
 	ast_assert(nodedata != NULL);
 	ast_assert(nodedatalength > 0);
 
-	/* setup the domain to lookup */
-	memset(domain, 0, sizeof(domain));
-	snprintf(domain, sizeof(domain), "%s.nodes.allstarlink.org", node);
+	/* AllStarLink supports two mechanisms to resolve node information.
+	 * You can use the SRV record followed by resolving the node name or
+	 * look up the information in the text record.
+	 */
+#if 1
+	/* Resolve the node by using SRV record */
+	{
+		char *hostname;
+		const char *ipaddress;
+		unsigned short iaxport;
+		
+		/* setup the domain to lookup */
+		memset(domain,0, sizeof(domain));
+		snprintf(domain, sizeof(domain), "_iax._udp.%s.nodes.allstarlink.org", node);
 
-	ast_debug(4, "Resolving DNS TXT records for: %s\n", domain);
+		ast_debug(4, "Resolving DNS SRV records for: %s\n", domain);
 
-	/* resolve the domain name */
-	if (ast_dns_resolve(domain, T_TXT, C_IN, &result)) {
-		ast_log(LOG_ERROR, "DNS request failed\n");
-		return -1;
-	}
-	if (!result) {
-		return -1;
-	}
+		if (ast_dns_resolve(domain, T_SRV, C_IN, &result)) {
+			ast_log(LOG_ERROR, "DNS SRV request failed\n");
+			return -1;
+		}
+		if (!result) {
+			ast_log(LOG_ERROR, "DNS SRV request failed\n");
+			return -1;
+		}
 
-	/* get the response */
-	record = ast_dns_result_get_records(result);
+		/* get the response */
+		record = ast_dns_result_get_records(result);
 	
-	if(!record) {
+		if(!record) {
+			ast_log(LOG_ERROR, "Resolving DNS did not return results for: %s\n", domain);
+			ast_dns_result_free(result);
+			return -1;
+		}
+
+		hostname = ast_strdupa(ast_dns_srv_get_host(record));
+		iaxport = ast_dns_srv_get_port(record);
+
+		ast_debug(4, "Resolving A record for host: %s, port: %d\n", hostname, iaxport);
+
 		ast_dns_result_free(result);
-		return -1;
+
+		if (ast_dns_resolve(hostname, T_A, C_IN, &result)) {
+			ast_log(LOG_ERROR, "DNS resolve request failed\n");
+			return -1;
+		}
+		if (!result) {
+			ast_log(LOG_ERROR, "Resolving A did not return results\n");
+			return -1;
+		}
+
+		/* get the response */
+		record = ast_dns_result_get_records(result);
+		if (!record) {
+			ast_log(LOG_ERROR, "Resolving A did not return results\n");
```

Comment:
Do you need a call to ast_dns_result_free(result) ?

Reply:
That was a problem.  Changes to be pushed.